ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/ProtocolSpecBuilder.java

```diff
@@ -118,11 +121,17 @@
   }
 
   public ProtocolSpecBuilder<T> precompileContractRegistryBuilder(
-      final Function<GasCalculator, PrecompileContractRegistry> precompileContractRegistryBuilder) {
+      final Function<PrecompiledContractConfiguration, PrecompileContractRegistry>
```

Comment:
This function can be changed such that you don't need appendPrivacyRegistry and you don't need to pass around so many functions of functions. When the 
```
  public ProtocolSpecBuilder<T> precompileContractRegistryBuilder(
      final Function<PrecompiledContractConfiguration, PrecompileContractRegistry>
          precompileContractRegistryBuilder) {
    this.precompileContractRegistryBuilder = (precompiledContractConfiguration) -> {
      final PrecompileContractRegistry registry = precompileContractRegistryBuilder.apply(precompiledContractConfiguration);
      if (precompiledContractConfiguration.getPrivacyParameters().isEnabled()) {
        registry.put(
          Address.DEFAULT_PRIVACY,
          new PrivacyPrecompiledContract(
            precompiledContractConfiguration.getGasCalculator(),
            precompiledContractConfiguration.getPrivacyParameters()
          )
        );
      }
      return registry;
    };
    return this;
  }
```